### DISCUSSION TITLE
ClearBrowsingDataAsync never returns

### DISCUSSION THREAD

**Discussion by karoulston9** [MicrosoftEdge/WebView2Feedback#3175] [Q&A]
Hello!

I would like some assistance in using WebView2. I have a WinForms app and on close I want to clear the profile cache. I await the call, but the clear never returns:

```
Private Async Function ClearBrowsing() As Task
        Dim profile As Microsoft.Web.WebView2.Core.CoreWebView2Profile = mBrowser.CoreWebView2.Profile
        '-- this never returns from await
        Await profile.ClearBrowsingDataAsync(CoreWebView2BrowsingDataKinds.DiskCache, Date.MinValue, Date.Now.AddDays(1))
        mBrowser.Dispose()
    End Function
```

Any suggestions? The mBrowser object is currently up and running when this is called, I tried disposing the web2 control first, but his destroys the profile object and I'm unable to call clear. This await will continue forever and clear nothing.

**Comment by novac42** [comment]
Hi @karoulston9 any exception thrown in the await call?
